File: app/src/main/java/com/example/tryanderror/NewsApp.java
package com.example.tryanderror;

import android.app.Application;
import android.content.SharedPreferences;
import androidx.appcompat.app.AppCompatDelegate;

public class NewsApp extends Application {
    @Override
    public void onCreate() {
        super.onCreate();

        SharedPreferences prefs = getSharedPreferences("news_prefs", MODE_PRIVATE);
        boolean isDarkMode = prefs.getBoolean("dark_mode", false);

        int mode = isDarkMode ? AppCompatDelegate.MODE_NIGHT_YES : AppCompatDelegate.MODE_NIGHT_NO;
        AppCompatDelegate.setDefaultNightMode(mode);
    }
}

File: app/src/main/java/com/example/tryanderror/NewsApp.kt
package com.example.tryanderror

import android.app.Application
import androidx.appcompat.app.AppCompatDelegate

class NewsApp : Application() {
    override fun onCreate() {
        super.onCreate()
        
        val prefs = getSharedPreferences("news_prefs", MODE_PRIVATE)
        val isDarkMode = prefs.getBoolean("dark_mode", false)
        
        val mode = if (isDarkMode) {
            AppCompatDelegate.MODE_NIGHT_YES
        } else {
            AppCompatDelegate.MODE_NIGHT_NO
        }
        AppCompatDelegate.setDefaultNightMode(mode)
    }
}
